src/project/manifest/pyproject.rs

```diff
@@ -0,0 +1,167 @@
+use pep508_rs::{Requirement, VersionOrUrl};
+use rattler_conda_types::{NamelessMatchSpec, PackageName, ParseStrictness::Lenient, VersionSpec};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::str::FromStr;
+use tokio::runtime::Handle;
+use toml_edit;
+use toml_edit::TomlError;
+
+use crate::pypi_name_mapping::conda_pypi_name_mapping;
+
+use super::{error::RequirementConversionError, ProjectManifest, SpecType};
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct PyProjectManifest {
+    #[serde(flatten)]
+    inner: pyproject_toml::PyProjectToml,
+    tool: Tool,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "kebab-case")]
+struct Tool {
+    pixi: ProjectManifest,
+}
+
+impl std::ops::Deref for PyProjectManifest {
+    type Target = pyproject_toml::PyProjectToml;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl PyProjectManifest {
+    /// Parses a toml string into a pyproject manifest.
+    pub fn from_toml_str(source: &str) -> Result<Self, TomlError> {
+        toml_edit::de::from_str(source).map_err(TomlError::from)
+    }
+}
+
+impl From<PyProjectManifest> for ProjectManifest {
+    fn from(item: PyProjectManifest) -> Self {
+        // Start by loading the data nested under "tool.pixi"
+        let mut manifest = item.tool.pixi.clone();
+
+        // TODO: tool.pixi.project.name should be made optional or read from project.name
```

Comment:
I would indeed use the `project.name` from the `pyproject.toml` here.

Reply:
@ruben-arts are you happy with making `ProjectMetadata.name` as optional, potentially checking the field is set in the `validate` method? Or should we go through an intermediary struct identical to `ProjectMetadata` but for an optional `name` field?

Reply:
I think it would be fine to error on not having a name for now.

Reply:
OK. I'll leave it as is then